### ISSUE TITLE
ST06 warning reported for pg_stat_activity without any wildcards selected

### ISSUE THREAD

**Issue by serpent7776** [sqlfluff/sqlfluff#6397]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

An ST06 lint warning is generated for this postgres query:

```
SELECT
    age(clock_timestamp(), query_start) AS duration,
    state
FROM
    pg_catalog.pg_stat_activity
```

### Expected Behaviour

No warnings reported, because there is no wildcards.

### Observed Behaviour

Warning is reported

```
L:   1 | P:   1 | ST06 | Select wildcards then simple targets before calculations
                       | and aggregates. [structure.column_order]
```

### How to reproduce

`sqlfluff lint --dialect postgres`

### Dialect

`postgres`

### Version

`sqlfluff, version 3.1.1, Python 3.12.6` installed from manjaro packages
`sqlfluff, version 3.2.4, Python 3.12.6` installed from pip

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rogalski** [comment]
FYI

Not a bug according to the rule specification.
Even without wildcards "_simple targets before calculations and aggregates_" is violated.

`age(...)` is the calculation, `state` is the simple target.

**Comment by serpent7776** [comment]
Oh, right. Now that you mention it I can see it. I guess I was too focused on the wildcards part. Sorry for the noise.

**Comment by WittierDinosaur** [comment]
Closing based on the above conversation 👍
